File: api/src/IChat.Api/Endpoints/Auth/V1/Mappings/AuthMapper.cs
namespace IChat.Api.Endpoints.Auth.V1.Mappings;

using IChat.Api.Endpoints.Auth.V1.DTOs;
using IChat.Core.Contracts.Auth;

public static class AuthMapper
{
    public static LoginRequest ToServiceRequest(this LoginDto dto) =>
        new()
        {
            UserName = dto.UserName,
            Password = dto.Password
        };

    public static UserResponse ToResponse(this UserView view) =>
        new()
        {
            Id = view.Id,
            UserName = view.UserName,
            DisplayName = view.DisplayName,
            Role = view.Role.ToString(),
            IsActive = view.IsActive,
            CreatedAt = view.CreatedAt
        };

    public static AuthResponse ToResponse(this AuthResult result) =>
        new()
        {
            AccessToken = result.AccessToken,
            ExpiresAt = result.AccessTokenExpiresAt,
            User = result.User.ToResponse()
        };
}

File: ui/src/app/core/auth/auth.mapper.ts
import type { AuthResponseDto, UserDto } from './auth.dto';
import type { AuthUser, Session } from './auth.model';

export function toAuthUser(dto: UserDto): AuthUser {
  return {
    id: dto.id,
    userName: dto.userName,
    displayName: dto.displayName,
    role: dto.role,
    isActive: dto.isActive,
    createdAt: dto.createdAt,
  };
}

export function toSession(dto: AuthResponseDto): Session {
  return {
    accessToken: dto.accessToken,
    expiresAt: dto.expiresAt,
    user: toAuthUser(dto.user),
  };
}
